File: backend/python/uploader/api/models/device.py
class Device():
    def __init__(self, model, serialNo, channels=[]):
        self.model = model
        self.serialNo = serialNo
        self.channels = channels
    
    def getModel(self):
        return self.model
    
    def getSerialNumber(self):
        return self.serialNo

    def getChannels(self):
        return self.channels

    def addChannel(self, channel):
        self.channels.append(channel)

    def clearChannels(self):
        self.channels = []

    def asDict(self):
        return {
            'model': self.model,
            'serialNo': self.serialNo,
            'channels': [ channel.asDict() for channel in self.channels ],
        }

File: backend/modules/uploader/api/models/device.js
class Device {
    constructor(model, serialNo, channels = []) {
        this.model = model;
        this.serialNo = serialNo;
        this.channels = channels;
    }

    getModel() {
        return this.model;
    }

    getSerialNumber() {
        return this.serialNo;
    }

    getChannels() {
        return this.channels;
    }

    addChannel(channel) {
        this.channels.push(channel);
    }

    clearChannels() {
        this.channels = [];
    }

    asObject() {
        return {
            model: this.model,
            serialNo: this.serialNo,
            channels: this.channels.map(channel => channel.asObject()),    
        };
    }
}

module.exports = {
    Device,
};